eth/celo_state_accessor.go

```diff
@@ -0,0 +1,35 @@
+package eth
+
+import (
+	"fmt"
+
+	"github.com/celo-org/celo-blockchain/core"
+	"github.com/celo-org/celo-blockchain/core/state"
+	"github.com/celo-org/celo-blockchain/core/types"
+)
+
+// Wraps `stateAtBlock` with the additional Celo-specific parameter afterNextRandomCommit.
+// This parameter executes the random commitment at the start of the next block,
+// since this is necessary for properly tracing transactions in the next block.
+func (eth *Ethereum) celoStateAtBlock(block *types.Block, reexec uint64, base *state.StateDB, checkLive bool, preferDisk bool, afterNextRandomCommit bool) (statedb *state.StateDB, err error) {
```

Comment:
Are we ever interested in the state without the randomness applied? At least in this PR it is always enabled, so we could remove that parameter.

Reply:
yes in `traceCall`, where the trace is applied to the end of the block (as opposed to the start of the following block), this one is used with "latest" often too, so would result in different behavior for the "latest" block vs. earlier blocks :-/ In this case I preferred keeping the behavior aligned with upstream but we can revisit!

Reply:
Ah, I looked at a partial diff, sorry.

Reply:
no worries!